File: 02/part2_answer.exs
Code.require_file "intcode_program.exs", __DIR__

{:ok, program_str} = File.read(__DIR__ <> "/input.txt")

noun = 12
verb = 2
program =
  IntcodeProgram.read_from(program_str)
  |> put_elem(1, noun)
  |> put_elem(2, verb)
  |> IntcodeProgram.run

IO.puts IntcodeProgram.output(program)

File: 02/part2_answer.rb
require_relative 'intcode_program'

input_file = "#{__dir__}/input.txt"
program_code = File.read(input_file)
desired_output = 19690720

(0..99).each do |input1|
  (0..99).each do |input2|
    program = IntcodeProgram.new(program_code, [input1, input2])
    program.run

    if program.output == desired_output
      puts 100 * input1 + input2
      break
    end
  end
end
